pkg/instance/service.go

```diff
@@ -97,6 +97,37 @@ func (s service) SaveInstance(instance *model.DeploymentInstance) error {
 	return s.instanceRepository.SaveInstance(instance)
 }
 
+func (s service) DeleteInstance(deploymentId, instanceId uint) error {
+	deployment, err := s.FindDeploymentById(deploymentId)
+	if err != nil {
+		return err
+	}
+
+	index := slices.IndexFunc(deployment.Instances, func(instance *model.DeploymentInstance) bool {
+		return instanceId == instance.ID
+	})
+	if index == -1 {
+		return errdef.NewNotFound("instance %d not found in deployment %d", instanceId, deployment.ID)
+	}
+	instance := deployment.Instances[index]
+
+	deployment.Instances = slices.DeleteFunc(deployment.Instances, func(instance *model.DeploymentInstance) bool {
+		return instanceId == instance.ID
+	})
+
+	_, err = s.validateNoCycles(deployment.Instances)
+	if err != nil {
+		return errdef.NewBadRequest("failed deleting instance: %v", err)
+	}
+
+	err = s.destroyDeploymentInstance(instance)
+	if err != nil {
+		return err
```

Comment:
Did you check the errors we get from `s.destroyDeploymentInstance` and below? I mostly see `return err` being returned.

Users should not get a k8s error (at least not as the start of the error message). So either `s.destroyDeploymentInstance` makes sure to return friendly errors or we `return fmt.Errorf("failed to destroy instance %d in deployment %d: %v", instanceId, deployment.ID,  err)` here.